azure/converters/tags_test.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/onsi/gomega"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+func Test_TagsToMap(t *testing.T) {
+	cases := []struct {
+		tags   infrav1.Tags
+		expect map[string]*string
+	}{
+		{
```

Comment:
let's add a bit more unit test cases to cover other scenarios when the src isn't nil as well

Reply:
maybe a good test would be to convert the value to map then back to tags and make sure that the value we get is strictly equal to the original value (ie. round trip conversion), since that's the bug we're fixing here.

Reply:
thank you for adding those!